Neuron implementation work

andrewgait opened the pull request:
Work to allow end users to modify neuron behaviour (specifically input_type, where inhibitory input could depend on excitatory input) more easily.

Depends on 
sPyNNaker pull request: SpiNNakerManchester/sPyNNaker#344
sPyNNaker8 pull request: https://github.com/SpiNNakerManchester/sPyNNaker8/pull/30

J-KC-Man commented:
Also updated my_threshold_type.py:

Updated def get_n_threshold_parameters(self) to return 2 parameters instead of 1 to prevent the following error when running example.py:

2017-02-26 00:50:42 ERROR: Error executing data specification for 0, 0, 2
Traceback (most recent call last):
…
data_specification.exceptions.DataSpecificationNoMoreException: Space unavailable to write all the elements requested by the write operation. Space available: 0; space requested: 4 for region 1.

andrewgait committed:
Neuron implementation work

- Add input_types and implementations to C code directories
- Add my_input_type and corresponding data holder in python
- Update examples file to run with my_input_type (NOTE: need to
play around with the numbers here so that it's different from
the basic my_neuron_model example)

andrewgait committed:
flake8 edits

andrewgait committed:
Add correct method to search for binaries

- noticed that this is what is being used by ExtraModelsPlugin,
so it should be used here too

rowleya commented on python_models/neuron/neuron_models/my_neuron_model.py:
```diff
@@ -85,7 +85,7 @@ def get_n_global_parameters(self):
         # Note: This must match the number of parameters in the global_neuron_t
         # data structure in the C code
-        return 1
+        return 3
```
Don't think this is right - there is only one global parameter that I can see

andrewgait replied:
I agree with @rowleya - there's only one global parameter here; there are 3 neural parameters, which is where the confusion may have arisen.

The other threshold_type error (there are 2 parameters in my_threshold_type, as Justin says) was fixed when we moved on from sPyNNakerNewModelTemplate to separate versions for PyNN 0.7 (sPyNNaker7NewModelTemplate) and PyNN 0.8 (sPyNNaker8NewModelTemplate).  We shouldn't be using sPyNNakerNewModelTemplate any more...

andrewgait committed:
Merge branch 'master' into neuron_implementation_work

rowleya commented:
Closing this - as Andrew points out, this is now fixed in another module.

andrewgait committed:
Tidy up comments

andrewgait committed:
Import correct quantities

andrewgait committed:
Add quantities to requirements list

rowleya committed:
Merge branch 'master' into neuron_implementation_work

Conflicts:
	setup.py

andrewgait committed:
Merge remote-tracking branch 'origin/master' into
neuron_implementation_work

Conflicts:
	examples/my_example.py
	requirements.txt

andrewgait committed:
Update to reflect sPyNNaker C code changes

andrewgait committed:
Update to match current sPyNNaker branch

andrewgait committed:
Move has_spiked into timestep_update

andrewgait committed:
Define new neuron model example

- use the changes in sPyNNaker (where input_type
and threshold_type can now be set to None) to 
define a model where all parameters are passed
into either the neuron_model or the synapse_type

andrewgait committed:
flake8

andrewgait committed:
flake8 again

andrewgait committed:
Actually use all defined parameters in new example

andrewgait committed:
Merge remote-tracking branch 'origin/master' into
neuron_implementation_work

Conflicts:
	c_models/src/neuron/Makefile
	c_models/src/neuron/builds/my_model_curr_exp/Makefile
	c_models/src/neuron/builds/my_model_curr_exp_my_additional_input/Makefile
	c_models/src/neuron/builds/my_model_curr_exp_my_threshold/Makefile
	c_models/src/neuron/builds/my_model_curr_exp_stdp_mad_my_timing_my_weight/Makefile
	c_models/src/neuron/builds/my_model_curr_my_synapse_type/Makefile

andrewgait committed:
Edit of classes to attempt to match master

- something still isn't quite right here...

andrewgait committed:
Merge remote-tracking branch 'origin/master' into
neuron_implementation_work

andrewgait committed:
Ignore debug build

andrewgait committed:
Flake8 remove unused import

andrewgait committed:
Edits to make all-in-one model work properly again

rowleya committed:
Merge branch 'master' into neuron_implementation_work

rowleya committed:
Fixed to match sPyNNaker

rowleya committed:
Standard models modified and building

rowleya committed:
Added full neuron and fixed things

andrewgait committed:
Fix unscaled machine time step

andrewgait committed:
Fixes to make my_example return correct answers

andrewgait committed:
Minor tidy up of figure from example script

andrewgait committed:
Do get_values in the correct order
